cocos/renderer/gfx-metal/MTLCommandBuffer.mm

```diff
@@ -507,24 +524,69 @@ of this software and associated engine source code (the "Software"), a limited,
         }
 
         if (sampler.stages & ShaderStageFlagBit::VERTEX) {
-            _commandEncoder.setVertexTexture(gpuDescriptor.texture->getMTLTexture(), sampler.textureBinding);
-            _commandEncoder.setVertexSampler(gpuDescriptor.sampler->getMTLSamplerState(), sampler.samplerBinding);
+            _renderEncoder.setVertexTexture(gpuDescriptor.texture->getMTLTexture(), sampler.textureBinding);
+            _renderEncoder.setVertexSampler(gpuDescriptor.sampler->getMTLSamplerState(), sampler.samplerBinding);
         }
 
         if (sampler.stages & ShaderStageFlagBit::FRAGMENT) {
-            _commandEncoder.setFragmentTexture(gpuDescriptor.texture->getMTLTexture(), sampler.textureBinding);
-            _commandEncoder.setFragmentSampler(gpuDescriptor.sampler->getMTLSamplerState(), sampler.samplerBinding);
+            _renderEncoder.setFragmentTexture(gpuDescriptor.texture->getMTLTexture(), sampler.textureBinding);
+            _renderEncoder.setFragmentSampler(gpuDescriptor.sampler->getMTLSamplerState(), sampler.samplerBinding);
+        }
+        
+        if(sampler.stages & ShaderStageFlagBit::COMPUTE) {
+            _computeEncoder.setTexture(gpuDescriptor.texture->getMTLTexture(), sampler.textureBinding);
         }
     }
 }
 
 void CCMTLCommandBuffer::blitTexture(Texture *srcTexture, Texture *dstTexture, const TextureBlit *regions, uint count, Filter filter) {
+    if(srcTexture && regions) {
+        id<MTLBlitCommandEncoder> encoder = [_mtlCommandBuffer blitCommandEncoder];
+        id<MTLTexture> src = static_cast<CCMTLTexture*>(srcTexture)->getMTLTexture();
+        id<MTLTexture> dst = nil;
+        CCMTLDevice* device = static_cast<CCMTLDevice*>(_device);
+        if(!dstTexture) {
+            id<CAMetalDrawable> currentDrawable = (id<CAMetalDrawable>)device->getCurrentDrawable();
+            dst = [currentDrawable texture];
```

Comment:
Is it designed like this? Change the content of onscreen texture with blit?

Reply:
before present() is called, it's still an offscreen texture, we haven't hold any pointer for presented texture. It's just a texture belongs to drawable, and it's legal to blit texture to drawable: https://developer.apple.com/forums/thread/63589

Reply:
Good to know, thanks.